src/main/java/org/sonar/plugins/stash/issue/MarkdownPrinter.java

```diff
@@ -120,12 +116,49 @@ private String formatTableList(String caption, List<String> items) {
     sb.append("|");
     sb.append(NEW_LINE);
     for (String item : items) {
-      sb.append("| ").append(item).append(" |").append(NEW_LINE);
+        sb.append("| ").append(item).append(" |").append(NEW_LINE);
+        if (this.includeFilesInOverview && issueToFilesMap.containsKey(item)) {
+            String files = String.join(", ", issueToFilesMap.get(item));
+            sb.append("| *Files: ").append(files).append("* |").append(NEW_LINE);
+        }
     }
 
     return sb.toString();
   }
 
+  private HashMap<String, ArrayList<String>> getMappedIssuesToFiles(Collection<PostJobIssue> allIssues) {
+    HashMap<String, ArrayList<String>> issueToFilesMap = new HashMap<>();
+    if (!this.includeFilesInOverview) {
+      return issueToFilesMap;
+    }
+
+    for (PostJobIssue issue : allIssues) {
+      String title = this.printIssueMarkdown(issue);
+      if (!issueToFilesMap.containsKey(title)) {
+        issueToFilesMap.put(title, new ArrayList<>());
+      }
+
+      if (issueToFilesMap.get(title).size() >= MAX_FILES_PER_ISSUE) {
+        continue;
+      }
+
+      Integer issueLine = issue.line();
+      if (issueLine == null) {
+        issueLine = 0;
+      }
+
+      issueToFilesMap.get(title).add(this.parseFileName(issue.componentKey(), issueLine));

```

Comment:
You should use the `IssuePathResolver` component to get access to the path of an issue.
This will properly handle multi-module projects.

Reply:
Also, I would prefer to sort the filename based on their length, shortest first (falling back to lexicographic order)
This gives them a stable order and will move the most important files to the front of the list.
(Assuming more important files have shorter names/are at the top of the file hierarchy)